src/frontend/js/widgets/Dashboard/utils/dashboardRoutes.tsx

```diff
@@ -0,0 +1,93 @@
+import { IntlShape, MessageDescriptor, useIntl } from 'react-intl';
+import type {
+  FormatXMLElementFn,
+  Options as IntlMessageFormatOptions,
+  PrimitiveType,
+} from 'intl-messageformat';
+import { Navigate, RouteObject } from 'react-router-dom';
+import { isEnumValue } from 'types/utils';
+import { DashboardLayoutRoute } from 'widgets/Dashboard/components/DashboardLayoutRoute';
+import {
+  LearnerDashboardPaths,
+  learnerDashboardRouteLabels,
+  learnerDashboardRoutePaths,
+} from 'widgets/Dashboard/utils/learnerRouteMessages';
+import { getLearnerDashboardRoutes } from 'widgets/Dashboard/utils/learnerRoutes';
+import { getTeacherDashboardRoutes } from 'widgets/Dashboard/utils/teacherRoutes';
+import {
+  TeacherDashboardPaths,
+  teacherDashboardRouteLabels,
+  teacherDashboardRoutePaths,
+} from './teacherRouteMessages';
+
+export interface DashboardRouteHandle {
+  crumbLabel?: MessageDescriptor;
+  renderLayout?: boolean;
+}
+
+export function getDashboardRoutes() {
+  const intl = useIntl();
+  const getRoutePath = getDashboardRoutePath(intl);
+  const routes: RouteObject[] = [
+    {
+      path: '/',
+      element: <DashboardLayoutRoute />,
+      children: [
+        {
+          index: true,
+          element: <Navigate to={getRoutePath(LearnerDashboardPaths.COURSES)} replace />,
+        },
+        ...getLearnerDashboardRoutes(),
+        {
+          path: getRoutePath(TeacherDashboardPaths.ROOT),
+          handle: {
+            renderLayout: true,
+          },
+          children: getTeacherDashboardRoutes(),
+        },
+      ],
+    },
+  ];
+  return routes;
+}
+
+/**
+ * Use `intl.formatMessage` to retrieve a path or label of a given route for the active locale.
+ *
+ * Currying function which takes first the type of attribute to translate (path or label)
+ * then takes the intl object retrieved from the IntlProvider.
+ * Finally, it takes a path and options to pass to `intl.formatMessage` method
+ */
+
+enum RouteAttributes {
+  PATH = 'path',
+  LABEL = 'label',
+}
+
+const getDashboardRouteAttribute =
```

Comment:
Ok so here is what I suggest. I find this approach more scalable and easy to maintain. WDYT all?


```
const DASHBOARD_ROUTE_LABELS = { ...learnerDashboardRouteLabels, ...teacherDashboardRouteLabels };
const DASHBOARD_ROUTE_PATHS = { ...learnerDashboardRoutePaths, ...teacherDashboardRoutePaths };
type DashboardPath = LearnerDashboardPaths | TeacherDashboardPaths;

const getDashboardRouteAttribute =
  (attribute: RouteAttributes) =>
  (intl: IntlShape) =>
  (
    path: DashboardPath,
    ...options: [
      values?: Record<string, PrimitiveType | FormatXMLElementFn<string, string>>,
      opts?: IntlMessageFormatOptions,
    ]
  ) => {
    const isLabel = attribute === RouteAttributes.LABEL;

    const message = isLabel ? DASHBOARD_ROUTE_LABELS[path] : DASHBOARD_ROUTE_PATHS[path];

    return intl.formatMessage(message, ...options);
  };

```

Reply:
+1 it's easy to read and simpler and as we are manipulating intl messages, we do not risk to override some keys with other.